File: src/models/entities.rs
use serde::{Serialize, Deserialize};
use sqlx::types::{Json};
use uuid::Uuid;

#[derive(Debug, Clone, Serialize, Deserialize, sqlx::FromRow)]
pub struct EntityModel {
    pub id: Uuid,
    pub ref_id: String,
    pub alt_id: Option<String>,
    pub name: String,
    pub active: Option<bool>,
}

#[derive(Debug, Clone, Serialize, Deserialize, sqlx::FromRow)]
pub struct Entity {
    pub id: Uuid,
    pub entity_model_id: Uuid,
    pub parent_id: Option<Uuid>,
    pub ref_id: String,
    pub alt_id: Option<String>,
    pub name: String,
    pub meta: Option<Json<serde_json::Value>>,
}

File: internal/models/entities.go
package models

import (
	"encoding/json"

	"github.com/google/uuid"
)

type EntityModel struct {
	ID     uuid.UUID `json:"id" db:"id"`
	RefID  string    `json:"ref_id" db:"ref_id"`
	AltID  *string   `json:"alt_id" db:"alt_id"`
	Name   string    `json:"name" db:"name"`
	Active *bool     `json:"active" db:"active"`
}

type Entity struct {
	ID            uuid.UUID        `json:"id" db:"id"`
	EntityModelID uuid.UUID        `json:"entity_model_id" db:"entity_model_id"`
	ParentID      *uuid.UUID       `json:"parent_id" db:"parent_id"`
	RefID         string           `json:"ref_id" db:"ref_id"`
	AltID         *string          `json:"alt_id" db:"alt_id"`
	Name          string           `json:"name" db:"name"`
	Meta          *json.RawMessage `json:"meta" db:"meta"`
}
